src/shared.js

```diff
@@ -0,0 +1,82 @@
+const request = require('request');
+
+class _SwarmJS {
+
+  constructor(opts) {
+    this.opts = opts || {};
+    if (this.opts.gateway) {
+      this.gateway = opts.gateway
+    } else if (this.opts.mode === 'http') {
+      this.gateway = 'http://swarm-gateways.net'
+    } else {
+      this.gateway = 'https://swarm-gateways.net'
```

Comment:
You have swarm-gateway twice. Maybe you can remove the `else if` then?

Reply:
Haha I knew it would trigger you 😜 I'll extract the `swarm-gateways.net`.